cime_config/stream_cdeps.py

```diff
@@ -427,7 +434,33 @@ def _days_in_month(month, year=1):
         next_month_start = datetime.date(next_year, next_month, 1)
         return (next_month_start - month_start).days
 
-    def _sub_paths(self, filenames, year_start, year_end):
+    @classmethod
+    def _add_day(cls, year, month, day):
```

Comment:
A couple of notes about this function:

(1) I considered using python's datetime module to do the addition of a day, but it seems like this would assume the use of a Gregorian (leap year) calendar, which we don't want here. Working around that felt like it would be at least as complicated as just writing our own function.

(2) In principle this could be extended to add N days instead of just one. But I'm having trouble envisioning a scenario where we'd want to add more than one day, so for now decided to keep it simple, following the YAGNI (You Aren't Gonna Need It) principle. We can always extend it later.